src/common/util/random.js

```diff
@@ -0,0 +1,113 @@
+/*globals define*/
+/*jshint node: true, browser: true, bitwise: false*/
+
+/**
+ * @author kecso / https://github.com/kecso
+ *
+ * collection of functions that uses random Numbers in WebGME
+ */
+
+define(['./chance.min'], function (ChanceJs) {
+    'use strict';
+
+    function random() {
+        var guidPool = '0123456789abcdef',
+            relidPool = '0123456789qwertyuiopasdfghjklzxcvbnmQWERTYUIOPASDFGHJKLZXCVBNM',
+            excludeList = ['atr', 'reg', 'ovr'],
+            maxTry = 2,
+            chance = new ChanceJs(),
+            randFunction = function (max) {
+                //return Math.floor(Math.random() * max);
+                return chance.natural({max:max-1});
+            },
+            validRelid = generateRelidRegexp();
+
+        function generateRelidRegexp() {
+            //(?!(^reg$)|(^atr$)|(^ovr$))^[0123456789regovat]+$
+            var regexp = '',
+                i;
+
+            //adding excludes
+            if (excludeList.length > 0) {
+                regexp += '(?!';
+                for (i = 0; i < excludeList.length; i += 1) {
+                    regexp += '(^' + excludeList[i] + '$)|';
+                }
+                regexp = regexp.slice(0, -1);
+                regexp += ')';
+            }
+            //adding the pool
+            regexp += '^(-)*[' + relidPool + ']+$';
+
+            return new RegExp(regexp);
+        }
+
+        function getRandomCharacter(pool) {
+            return pool.charAt(randFunction(pool.length));
+        }
+
+        function generateGuid() {
+            var S4 = function () {
+                return getRandomCharacter(guidPool) +
+                    getRandomCharacter(guidPool) +
+                    getRandomCharacter(guidPool) +
+                    getRandomCharacter(guidPool);
+            };
+
+            return (S4() + S4() + '-' + S4() + '-' + S4() + '-' + S4() + '-' + S4() + S4() + S4());
+        }
+
+        function generateRelid(object) {
+            var relid,
+                i,
+                length = 1,
+                tries = 0;
+
+            do {
+                if (tries >= maxTry) {
+                    tries = 0;
+                    length += 1;
+                }
+                relid = '';
+                for (i = 0; i < length; i += 1) {
+                    relid += getRandomCharacter(relidPool);
+                }
+                tries += 1;
+            } while (object[relid] !== undefined);
```

Comment:
Would hasOwnProperty work? Or are there cases where the object gets assigned an undefined?